### DISCUSSION TITLE
Using RUM in OpenObserve Docker Image

### DISCUSSION THREAD

**Discussion by APT256** [openobserve/openobserve#3285] [General]
I am trying to find documentation for setting up RUM to work on an OpenObserve Docker image. Does anyone know if this has been done?

**Comment by hengfeiyang** [comment]
In the version > 0.8.x already have the RUM feature.

**Reply by APT256** [reply]
My bad, I shouthave been more specific. I found the instructions for setting up RUM, but they appear to be for a stand-alone installation instead of a docker installation as I can not locate the configuration files mentioned in the RUM setup documentation.

**Reply by bjp232004** [reply]
Allow me to clarify. You might be following similar steps, but just to ensure we're on the same page, this configuration is meant for your front-end application. Each option's value will be substituted with an environment variable, which you can specify in your Docker configuration.

If you're moving forward with this approach, we can arrange a quick call to delve into it further.

**Reply by APT256** [reply]
@bjp232004 thanks for following up. Let me give you a little background of my environment (it might help). I am running this in a lab that it hosted off of a Unraid server (everything is either virtual or docker). OpenObserve was installed from the Unraid App Store and is running as a docker container. All my servers are VMs. When I go into the web interface of OpenObserve, select "Ingestion", select "Real User Monitoring", I see a page that states the following: 

"Step1: Add [@openobserve/browser-rum](https://www.npmjs.com/package/@openobserve/browser-rum) and [@openobserve/browser-logs](https://www.npmjs.com/package/@openobserve/browser-logs) to your package.json file, or run the following command:

npm i @openobserve/browser-rum @openobserve/browser-logs


Step2: Initialize the OpenObserve RUM and Logs SDKs in your application entry point (e.g. index.js or main.js).


import { openobserveRum } from '@openobserve/browser-rum';
import { openobserveLogs } from '@openobserve/browser-logs';

const options = {
  clientToken: 'rum2LrZDTPaflTu8KFZ',
  applicationId: 'web-application-id',
  site: '192.168.100.254:5080',
  service: 'my-web-application',
  env: 'production',
  version: '0.0.1',
  organizationIdentifier: 'default',
  insecureHTTP: true,
  apiVersion: 'v1',
};

openobserveRum.init({
  applicationId: options.applicationId, // required, any string identifying your application
  clientToken: options.clientToken,
  site: options.site,
  organizationIdentifier: options.organizationIdentifier,
  service: options.service,
  env: options.env,
  version: options.version,
  trackResources: true,
  trackLongTasks: true,
  trackUserInteractions: true,
  apiVersion: options.apiVersion,
  insecureHTTP: options.insecureHTTP,
  defaultPrivacyLevel: 'allow' // 'allow' or 'mask-user-input' or 'mask'. Use one of the 3 values.
});

openobserveLogs.init({
  clientToken: options.clientToken,
  site: options.site,
  organizationIdentifier: options.organizationIdentifier,
  service: options.service,
  env: options.env,
  version: options.version,
  forwardErrorsToLogs: true,
  insecureHTTP: options.insecureHTTP,
  apiVersion: options.apiVersion,
});

// You can set a user context
openobserveRum.setUser({
  id: "1",
  name: "Captain Hook",
  email: "captainhook@example.com",
});

openobserveRum.startSessionReplayRecording();"

I am guessing that the package.json file needs to be on persistent storage, so I would assume it should be located at: \mnt\user\appdata\openobserve as this is the only persistent data storage mapped to openobserve. Now, if the package.json file is actually used to "build" the openobserve container, then I guess I have other issues as I am using the Unraid App Store, and I am not sure how I would go about changing that file in an App Store installer. If this is the case, I will try to install it on a lab Docker server that is strictly Docker.
